src/protocols/internal/encode.c

```diff
@@ -57,38 +57,31 @@ static ssize_t internal_encode(fr_dbuff_t *dbuff,
 			       fr_da_stack_t *da_stack, unsigned int depth,
 			       fr_cursor_t *cursor, void *encoder_ctx)
 {
-	uint8_t				*enc_field, *len_field, *value_field, *value_end;
-	fr_dict_attr_t const		*da = da_stack->da[depth];
-	VALUE_PAIR			*vp = fr_cursor_current(cursor);
+	fr_dbuff_t		work_dbuff = FR_DBUFF_NO_ADVANCE(dbuff);
+	uint8_t			*enc, *data;
+	fr_dict_attr_t const	*da = da_stack->da[depth];
+	VALUE_PAIR		*vp = fr_cursor_current(cursor);
 
-	uint8_t				flen;
-	ssize_t				slen = 0;
+	uint8_t			flen;
+	ssize_t			slen;
 
-	uint8_t				buff[sizeof(uint64_t)];
+	uint8_t			buff[sizeof(uint64_t)];
 
 	FR_PROTO_STACK_PRINT(da_stack, depth);
 
-	enc_field = dbuff->p;
+	enc = work_dbuff.p;
 
 	/*
 	 *	Zero out first encoding byte
 	 */
-	fr_dbuff_memset(dbuff, 0, 1);
```

Comment:
Why was this removed?

Reply:
Looking at the code, it's now `FR_DBUFF_BYTES_IN_RETURN(&work_dbuff, 0);`. If that's not there, please let me know.

Reply:
Ah, ok, apologies, missed that.